flow/packet-headers/ipv6_routing.yaml

```diff
@@ -0,0 +1,397 @@
+              is 2. When AUTO is assigned the value is set to one less than the number 
+              of segment specified.
+            format: integer
+            length: 8
+            default: 0
+            features: [auto, count]
+          x-field-uid: 2
+        flags:
+          $ref: '#/components/schemas/Flow.Ipv6SegmentRouting.Flags'
+          x-field-uid: 3
+        tag:
+          x-field-pattern:
+            description: 16-bit field used to tag a packet as part of a class or group,
+              enabling shared properties or policy treatment. If not used, it MUST
+              be set to zero.
+            format: integer
+            length: 16
+            default: 0
+            features:
+              - count
+          x-field-uid: 4
+        segment_list:
+          description: 128-bit IPv6 addresses representing the nth segment in the
+            Segment List. The Segment List is encoded starting from the last segment
+            of the SR Policy. That is, the first element of the Segment List (Segment List[0])
+            contains the last segment of the SR Policy, the second element contains
+            the penultimate segment of the SR Policy, and so on.
+          type: array
+          minItems: 1
+          items:
+            $ref: '#/components/schemas/Flow.Ipv6SegmentRouting.Segment'
+          x-field-uid: 5
+        tlvs:
+          type: array
+          items:
+            $ref: '#/components/schemas/Flow.Ipv6SegmentRouting.Tlvs'
+          x-field-uid: 6
+    Flow.Ipv6SegmentRouting.Segment:
+      type: object
+      properties:
+        segment:
+          x-field-pattern:
+            description: A 128-bit IPv6 address segment.
+            format: ipv6
+            default: ::0
+            features:
+              - count
+          x-field-uid: 1
+    Flow.Ipv6SegmentRouting.Tlvs:
+      description: A list of optional Type-Length-Value (TLV) objects that carry
+        metadata for segment processing. TLVs are only present if the Hdr Ext
+        Len is greater than what is required for the fixed header and the segment
+        list.
+      type: object
+      properties:
+        choice:
+          type: string
+          default: ingress
+          x-field-uid: 1
+          x-enum:
+            ingress:
+              x-field-uid: 1
+            egress:
+              x-field-uid: 2
+            opaque:
+              x-field-uid: 3
+            padding:
+              x-field-uid: 4
+            path_trace:
+              x-field-uid: 5
+        type:
+          x-field-pattern:
+            description: Type identifier for the TLV. AUTO will set it to correct value
+              based on choice.
+            format: integer
+            length: 8
+            default: 0
+            features: [auto, count]
+          x-field-uid: 2
+        length:
+          x-field-pattern:
+            description: The length of the TLV value in bytes.
```

Comment:
If auto is selected , the implementation should automatically set the length field to the correct length as per the selected TLV and attributes,

Reply:
Done.